pkg/scheduler/plugins/coscheduling/core/core.go

```diff
@@ -283,8 +283,9 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, state *framework.Cy
 		return fmt.Errorf("gang has not init, gangName: %v, podName: %v", gang.Name,
 			util.GetId(pod.Namespace, pod.Name))
 	}
+
 	// resourceSatisfied means pod will directly pass the PreFilter
-	if gang.getGangMatchPolicy() == extension.GangMatchPolicyOnceSatisfied && gang.isGangOnceResourceSatisfied() {
+	if gang.getGangMatchPolicy() == extension.GangMatchPolicyOnceSatisfied && gang.isGangOnceResourceSatisfied() && !gang.CloseHistoryEvaluate {
```

Comment:
what is the difference between the CloseHistoryEvaluate and the GangMatchPolicyWaitingAndRunning?

Reply:
Can GangMatchPolicyWaitingAndRunning solve the above problem?

Reply:
FYI: https://github.com/koordinator-sh/koordinator/blob/main/docs/proposals/scheduling/20220901-gang-scheduling.md#gang-match-policy